pkg/operation/botanist/health_check.go

```diff
@@ -816,6 +813,31 @@ func (b *Botanist) healthChecks(initializeShootClients func() error, thresholdMa
 	return apiserverAvailability, controlPlane, nodes, systemComponents
 }
 
+/*This function is created because when Loki is deployed for the first time on already existing shoot
+the maintenance window will not allow Loki StatfulSet to be deployed and the health check will fail.
+Do we are preventing this health check up to a certain version when Loki will be still integrated to the
+landscapes. In versions above 1.8.0 this function can be removed */
```

Comment:
```suggestion
/*Shoot clusters prior gardener v1.8 were using EFK based stack, but from this version onward Loki is used.
On some landscapes, shoots clusters are reconciled only in their maintenance time window and to prevent failing health checks, they are executed only for shoots that have already been reconciled by Gardener v1.8.x+ */
```